### ISSUE TITLE
[Question] locale: "de-DE" not working in firefox and webkit, am I missing something?

### ISSUE THREAD

**Issue by Quietscheentchen13** [microsoft/playwright#17181]
Hi,

I defined locale: "de-DE" in our projects config file. For chromium everything works fine. In firefox and webikit, this setting seems to have not the desired effect. I am having problems with expected date/time formats in inputs.

Do I miss something or am I doing something wrong? Is there more that has to be configured? I checked the Docs, but without success.

Thanks in advance for any help.

**Comment by dgozman** [comment]
@Quietscheentchen13 Locale is supposed to work in all browsers. Could you share a code snippet that does not work as expected? This way we'll be able to understand and fix the issue.

**Comment by Quietscheentchen13** [comment]
@dgozman Thanks for your fast reply.

Sorry, I am trying to explain my problem more precise and share the code of my config:
I have some inputs where I have to enter a defined date and a defined time. When I run a test in firefox for example, ff starts with the language set to english. The inputs are now expecting an english time format (with am, pm). When I keep the browser open after the test, change the language to german and revisit the site, the inputs are expecting the german time format. So the browser language is affecting the expected date/time format for the inputs. I thought locale would cover that.
For chromium everything works fine.

Example:
I have a time defined (as string) "16:45:30" (HH:mm:ss) which I want to enter in the input field.
Running test in ff (language is english by default): Expected time format of the input is (hh:mm:ss am/pm) => Test will fail later
Keeping ff open and changing the language to german: Expected time format of the input is (HH:mm:ss) => Test will pass later

Heres the content of my playwright.config.ts:

```ts
import type { PlaywrightTestConfig } from "@playwright/test";
import { devices } from "@playwright/test";

const config: PlaywrightTestConfig = {
  testDir: "./tests",
  timeout: 2 * 60 * 1000,
  expect: {
    timeout: 5 * 1000,
  },
  retries: 1,
  workers: 3,
  reporter: [["list"], ["html", { open: "on-failure" }]],
  use: {
    baseURL: "https://exampleurl.com",
    screenshot: "only-on-failure",
    trace: "on-first-retry",
    locale: "de-DE",
    actionTimeout: 20 * 1000,
    navigationTimeout: 30 * 1000,
  },

  projects: [
    {
      name: "smoketestChromium",
      testDir: "./tests/smoketest",
      retries: 1,
      use: {
        browserName: "chromium",
        viewport: { width: 1920, height: 1080 },
      },
    },

    {
      name: "smoketestFirefox",
      testDir: "./tests/smoketest",
      testIgnore: /1999_Smoketest Login.spec.ts/,
      retries: 1,
      use: {
        browserName: "firefox",
        viewport: { width: 1920, height: 1080 },
      },
    },

    {
      name: "smoketestWebkit",
      testDir: "./tests/smoketest",
      testIgnore: /1999_Smoketest Login.spec.ts/,
      retries: 1,
      use: {
        browserName: "webkit",
        viewport: { width: 1920, height: 1080 },
      },
    },
};

export default config;
```

**Comment by dgozman** [comment]
@Quietscheentchen13 Is it possible to share some code from the page, or the page url? We have some tests for locale and date formatting, that coincidentally [happen to use `de-DE`](https://github.com/microsoft/playwright/blob/main/tests/library/browsercontext-locale.spec.ts#L65-L70) and they work as expected. Perhaps we have missed some edge case, so having a particular snippet from the page that behaves incorrectly would be very helpful.

**Comment by Quietscheentchen13** [comment]
@dgozman
I'm sorry, but unfortunately I'm not able to share any code from the page or the page url.

I checked a site that returns me the locale (https://www.localeplanet.com/support/browser.html).
Here the results, when firefox is set to english (via browser settings):
```
navigator.browserLanguage: undefined
navigator.language: de-DE
date.toLocaleDateString(): 9/12/2022
date.toLocaleTimeString(): 8:00:38 AM
```

Here for firefox set to german (via browser settings):
```
navigator.browserLanguage: undefined
navigator.language: de-DE
date.toLocaleDateString(): 12.9.2022
date.toLocaleTimeString(): 08:03:46
```
I really have not much knowledge about this topic and I don't know, if this site with the results is trustworty.
But it's interesting, that the results seem to show exactly the behaviour of the date/time input fields.
Changing the browser language is not affecting the locale (which in the results is correct: de-DE), but the date and time format.

Is there maybe a way to define the browser language in the config? Or am I on the wrong track?

**Comment by Quietscheentchen13** [comment]
@dgozman

The firefox version used by playwright is Nightly 103.0.
_Is this correct? What would be the correct way to update the browser installs?_

German language is not in the settings by default. It has to be added.
_Could that cause problems? That it's not available for the browser to choose from?_

**Comment by mxschmitt** [comment]
@Quietscheentchen13 we need a small repro / code which we can run locally. Or know which Web APIs your site is using otherwise we can't act on it. As of now it looks like everything is behaving how it should be.

**Comment by Quietscheentchen13** [comment]
@mxschmitt 
I'm very sorry, but it's really not possible to share any site code.
We did some more research and I have to agree, that on the playwrigth side, everything seems ok. The request and response headers show the configured locale.
As you already mentioned it's a problem of the site. The date/time fields somehow use the browser language as a reference for the expected date/time format and not the locale.

The link I posted here earlier showed a similar behavior. Where the date/time format is changing based on the browser language. Maybe that's something to look at.

I will close this issue and thank you all for your patience and help.
